docs/design/proposals/labels.md

```diff
@@ -0,0 +1,109 @@
+
+#### Design details
+
+We will rename or introduce the following labels, which will be defined in `@pkg/metadata/labels.go`:
+
+1.  **Labels for mapping created resources back to the instance:**
+    - To provide a direct and queryable link from a created resource back to the instance that caused its creation. This set of labels uniquely identifies the instance.
+    - **Proposed Labels:**
+      - `kro.run/instance-gvk: mygroup.example.com/v1/MyKind`: The Group Version Kind (GVK) of the instance
+      - `kro.run/instance: default/my-instance`: The namespace and name of the instance.
+      - `app.kubernetes.io/instance: MyKind/default/my-instance`: Kind namespace and name of the instance
+    - **Existing Labels**
+      - `kro.run/instance-id: 6e6a1d95-4855-4062-b51b-8f288cb96365`
+      - `kro.run/instance-name: test-instance`
+      - `kro.run/instance-namespace: default`
+2.  **Label for mapping created resources back to the RGD:**
+    - To identify which RGD was used as a template to create a resource during the reconciliation of an instance. This label will be applied to all resources created by the instance controller.
+    - **Proposed Labels:**
+      - `kro.run/part-of: my-rgd-name` - The name of the ResourceGraphDefinition
+      - `app.kubernetes.io/part-of: my-rgd-name`
+    - **Existing Labels:**
+      - `kro.run/resource-graph-definition-id: 761e8fb7-14d5-4da1-b4f9-0a16fc334d6c`
+      - `kro.run/resource-graph-definition-name: my-rgd-name`
+3. **Label for nested RGD instance:**
+    - To differenciate the RGD that defines (and hence reconciles) the instance and the RGD that created it, we need to introduce a separate label.
+    - Currently we dont differentiate, it results in a conflict (https://github.com/kro-run/kro/pull/631)
+    - To indicate that an instance is being actively reconciled against a specific RGD. This label will be applied to the instance itself.
+    - **Proposed Labels:**
+      - `kro.run/managed-by-rgd: my-rgd-name` - name of the ResourceGraphDefinition that reconciles the instance.
+    - **Existing Labels:**
+      - `kro.run/resource-graph-definition-id: 761e8fb7-14d5-4da1-b4f9-0a16fc334d6c`
+      - `kro.run/resource-graph-definition-name: my-rgd-name`
+      - Conflicts with label used in creation path for nested RGD (https://github.com/kro-run/kro/pull/631)
+4. **Kubernetes Common Labels:**
+    - **Purpose:** Conforming to standard/common k8s labels allows easier integration across tools in k8s ecosystem.
+    - **Proposed Labels:**
+      - `app.kubernetes.io/managed-by: KRO` : Fixed value
+
+## Other solutions considered
+
+1. We could continue using the existing labels, but their purpose is not as explicit for relationship tracking, which can lead to confusion for users and client tools.
+2. We could also use annotations, but labels are better suited for this purpose as they are queryable via the Kubernetes API, which is a key requirement for observability and tooling.
```

Comment:
I would suggest 1 or 2 annotations for programmatic access that include IDs

for example


```
instance.kro.run/reference: {"apiVersion": "kro.run/v1alpha1", "kind": "MyKind", "resource": "mykinds", "namespace": "default", "name": "my-instance", "id": "$id"}
rgd.kro.run/reference: {"apiVersion": "kro.run/v1", "name": "my-rgd", "id": "${id}"}
```

ID would help reconcilers detect that the instance (or RGD) was fully replaced between 2 reconciliation loops